text/0001-PXE-Booting-UEFI.md

```diff
@@ -62,19 +62,24 @@ Foreman Proxy must be enhanced with new PXE variant "Pxegrub2" to allow deployin
 
 ### [Foreman core #12634](http://projects.theforeman.org/issues/12634) - New HW Model flags pxe_template and loader
 
-Hardware model will be enhanced with two new flags: PXE template specifies PXE template kind and variant pair to use. Until today, this was hardcoded to PXELinux or PXEGrub by selected operating system. Optionally, template variant flag from OS could be simply dropped because PXE template implies variant to use and there are no variants which share the same configuration syntax (this is an open item).
+Foreman will support both BIOS and UEFI systems and it is required to associate extra information for each indidivual host (or hostgroup) in order to provide TFTP "filename" option correctly. It can be set per instance via new Host flag (free-form dropdown field) called "PXE Loader" (`pxe_loader` model attribute) with possible values:
 
-Orchestration TFTP code must be modified to use multiple "KIND LOADER default local boot" and "KIND LOADER global default" templates. Local and global boot templates must be added for Grub2 new kind. Naming convention will be:
+* pxelinux.0 (default value)
```

Comment:
I'm wondering if we should do the following to avoid complicating the life of the user more (I don't think users need to be concerned about which file is used to boot their hosts):

* Start by serving the PXE template with filename  'pxegrub'. If this works,  shortly after we will get a request for the `provision` kind of template. End of the story.
* If it doesn't work, serve the template again with filename 'grub/bootx64.efi'. Rinse and repeat until we find a filename that works.

This assumes the computer will reboot itself after some time if the PXE template wasn't suitable, or Foreman itself can reboot it if it doesn't ask for a 'provision' template in a reasonable amount of time. I realize this can make host creation slower, so maybe we can set this flag (optional) too but not make it mandatory?

Does that sound good @lzap ? My main goal with this is to not complicate the host form further.